Modules/StudyProgramme/classes/class.ilPrgRestartAssignmentsCronJob.php

```diff
@@ -103,21 +110,103 @@ public function getDefaultScheduleValue()
     }
         
     /**
-     * Run job
-     *
      * @return ilCronJobResult
      */
     public function run()
     {
         $result = new ilCronJobResult();
-        $result->setStatus(ilCronJobResult::STATUS_OK);
-        foreach ($this->user_assignments_db->getDueToRestartInstances() as $assignment) {
-            try {
-                $assignment->restartAssignment();
-            } catch (ilException $e) {
-                $this->log->write('an error occured: ' . $e->getMessage());
+        $result->setStatus(ilCronJobResult::STATUS_NO_ACTION);
+
+        $programmes_to_reassign = $this->getSettingsRepository()
+            ->getProgrammeIdsWithReassignmentForExpiringValidity();
+
+        if (count($programmes_to_reassign) == 0) {
+            return $result;
+        }
+
+        $today = $this->getNow();
+        $programmes_and_due = [];
+
+        foreach ($programmes_to_reassign as $programme_obj_id => $days_offset) {
+            $interval = new DateInterval('P' . $days_offset . 'D');
+            $due = $today->add($interval);
+            $programmes_and_due[$programme_obj_id] = $due;
+        }
+
+        //TODO: expire for assignment, not progress!!!
+        $progresses = $this->getProgressRepository()
+            ->getAboutToExpire($programmes_and_due, false);
+
+        if (count($progresses) == 0) {
+            return $result;
+        }
+    
+        $events = $this->getEvents();
+        $assignment_repo = $this->getAssignmentRepository();
+        foreach ($progresses as $progress) {
+            $ass = $assignment_repo->read($progress->getAssignmentId());
```

Comment:
```suggestion
            $ass = $assignment_repo->get($progress->getAssignmentId());
```
Same wording thing as above: Just give us that assignment, don't care if it was "read" somehow.